actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -49,6 +49,21 @@ class TagBuilder # :nodoc:
         HTML_VOID_ELEMENTS = %i(area base br col embed hr img input keygen link meta param source track wbr).to_set
         SVG_SELF_CLOSING_ELEMENTS = %i(animate animateMotion animateTransform circle ellipse line path polygon polyline rect set stop use view).to_set
 
+        def self.define_void_element(name, self_closing: false)
+          class_eval <<~RUBY, __FILE__, __LINE__ + 1
+            def #{name}(content = nil, escape: true, **options, &block)
```

Comment:
@rafaelfranca Related to your comment on https://github.com/rails/rails/pull/49368#discussion_r1336579024:

What can we do to hint to RDoc that these methods are being defined?

The [RDoc documentation on metaprogramming](https://github.com/ruby/rdoc/blob/cac17095c36ffe90f327b880ac9e0ac9819a4365/lib/rdoc/parser/ruby.rb#L83-L142) relies on the list of methods being known ahead of time. Since these are being defined in a loop, I'm not sure how to communicate with RDoc.

Reply:
Oh, yeah. I think there is no way to do this other than we explicitly adding all the `:method: some_method!` definitions to the doc. It will be a big list and we can't automate, but at least they don't change often.